src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -394,12 +394,14 @@ def __init__(
         order_by: Optional[List[Expression]] = None,
         limit_: Optional[int] = None,
         offset: Optional[int] = None,
+        qualify: Optional[Expression] = None,
```

Comment:
It's interesting that we have `qualify` here but not `having` 🤔

Reply:
True. There is no active demand for these SQL features that don't have a direct counterpart in Spark. It shouldn't be much work to support it tho.